server/blob/blob-objectstorage/src/main/java/org/apache/james/blob/objectstorage/ObjectStorageBlobStore.java

```diff
@@ -211,6 +223,7 @@ public PayloadCodec getPayloadCodec() {
     public Mono<Void> delete(BucketName bucketName, BlobId blobId) {
         ObjectStorageBucketName resolvedBucketName = bucketNameResolver.resolve(bucketName);
         return Mono.<Void>fromRunnable(() -> blobStore.removeBlob(resolvedBucketName.asString(), blobId.asString()))
+            .then(blobExistenceTester.delete(resolvedBucketName, blobId))
```

Comment:
There is a short timewindow between blob deletion & cache cleanup.

I suggest cleaning up the cache first.

Reply:
I don't get why we need to delete from the existence table twice, could you explain?